theano/tensor/opt.py

```diff
@@ -3923,27 +3923,32 @@ def local_elemwise_sub_zeros(node):
 
 @register_canonicalize
 @register_specialize
-@gof.local_optimizer([T.Sum])
-def local_sum_div_dimshuffle(node):
+@gof.local_optimizer([T.Sum, T.elemwise.Prod])
+def local_sum_prod_div_dimshuffle(node):
     '''sum(a / dimshuffle{...}(b), axis=l) -> sum(a, axis={...}) / b,
-    if dimension l of the DimShuffle is 'x'.'''
-    # TODO: extend it to product, and quotient of products
+       if dimension l of the DimShuffle is 'x'
+
+       or
+
+       prod(a / dimshuffle{...}(b), axis=l) -> prod(a, axis={...}) / b,
+      if dimension l of the DimShuffle is 'x'
+    '''
 
     # It does not make much sense now to extend it to the case where the
     # dimshuffle is in the numerator, since elemwise inversion of the
-    # denominator would still be needed before the summation.
+    # denominator would still be needed before the summation or production.
 
-    if isinstance(node.op, T.Sum):
+    if isinstance(node.op, (T.SumT.elemwise.Prod)):
```

Comment:
Sorry for this mistake. I resolved it.

On Fri, Jul 17, 2015 at 10:36 AM, carriepl <notifications@github.com> wrote:

> In theano/tensor/opt.py
> <https://github.com/Theano/Theano/pull/3149#discussion_r34894529>:
>
> >
> > -    if isinstance(node.op, T.Sum):
> > +    if isinstance(node.op, (T.SumT.elemwise.Prod)):
>
> You need a comma here between T.Sum and T.elemwise.Prod
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/3149/files#r34894529>.
>